### DISCUSSION TITLE
How to add custom icon before the menu hyperlinks( for .md file routelinks)

### DISCUSSION THREAD

**Discussion by Kishore2121** [squidfunk/mkdocs-material#5288] [Q&A] [answered]
@squidfunk and anyone,

can you please tell me how to add custom icon like below for menu items. Appreciate your help in advance.

![image](https://user-images.githubusercontent.com/24959357/228324304-2a7fe6fe-16f7-42aa-ba6e-300d01d7be4a.png)

Thanks

**Comment by squidfunk** [comment, marked as answer]
Jup, if you're using [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), you can set a [page icon](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-icon):

``` yaml
---
icon: material/emoticon-happy 
---

# Document title
...
```

**Reply by Kishore2121** [reply]
Hi @squidfunk , thanks for your reply.
I'm not using the material insiders. I'm using the theme named material and I have **nav** items like below and beside each items I need a book icon like in the attached image.

nav:
  - legal information: Legal_info/Legal_Information.md
  - User Manual:  Legal_info/User_Manual.md

**Reply by squidfunk** [reply]
As said, it's only possible with Insiders right now. When the [next funding goal](https://squidfunk.github.io/mkdocs-material/insiders/#12000-piri-piri) is hit, this feature will be released in the community edition as well. Until then, you might consider using emojis in your titles, or implement the functionality yourself.
